madmom/audio/cepstrogram.py

```diff
@@ -222,6 +245,63 @@ def __new__(cls, spectrogram, filterbank=MelFilterbank,
         # return the object
         return obj
 
+    @staticmethod
+    def calc_deltas(data, delta_filter):
+        """
+        Applies the given filter to the data after automatically padding by
+        replicating the first and last frame. The length of the padding is
+        calculated via ceil(len(delta_filter))
+        :param data: a numpy array with the data
+        :param delta_filter: a numpy array containing the filter
+        :return: a numpy array of the same shape as data, containing the deltas
+        """
+        # prepare vectorized convolve function
+        # (requires transposed matrices in our use case)
+        vconv = np.vectorize(partial(np.convolve, mode="same"),
+                             signature='(n),(m)->(k)')
+        # pad data by replicating the first and the last frame
+        k = int(math.ceil(len(delta_filter) / 2))
+        padded = np.vstack((np.array([data[0], ] * k),
+                            data,
+                            np.array([data[-1], ] * k)))
+        # calculate the deltas for each coefficient
+        deltas = vconv(padded.transpose(), delta_filter)
+        return deltas.transpose()[k:-k]
+
+    @lazyprop
+    def deltas(self, delta_filter=MFCC_DELTA_FILTER):
+        return MFCC.calc_deltas(self, delta_filter)
+
+    @lazyprop
+    def deltadeltas(self, deltadelta_filter=MFCC_DELTADELTA_FILTER):
+        return MFCC.calc_deltas(self.deltas, deltadelta_filter)
+
+    def calc_ml_deltas(self, delta_filter=MFCC_DELTA_FILTER,
```

Comment:
I kind of dislike the abbreviation "ml" for Matlab, since it is used for machine learning elsewhere in the package.